### PR TITLE
Update to TypeSpec 0.55.0

### PR DESCRIPTION
**Pull request by timotheeguerin**

### CODE REVIEW

File: eng/pipelines/typespec-validation.yml
Comment (mikeharder) [inline eng/pipelines/typespec-validation.yml:1]: Revert this file

File: package.json [hunk 1]
```diff
@@ -1,18 +1,17 @@
 {
   "name": "azure-rest-api-specs",
   "devDependencies": {
-    "@azure-tools/typespec-apiview": "0.4.5",
```
Comment (mikeharder) [inline package.json:4]: Add back `typespec-apiview` with the same version

**Review by mikeharder — CHANGES_REQUESTED** [review]
Revert changes to typespec-validation.yml and add back typespec-apiview to package.json

**Review by mikeharder — COMMENTED** [review]
Approving from the perspective of EngSys, but failing checks like "Swagger BreakingChanges" should be approved by the correct people.

**Review by mikeharder — APPROVED** [review]

**Review by markcowl — APPROVED** [review]

File: specification/ai/data-plane/DocumentIntelligence/preview/2024-02-29-preview/DocumentIntelligence.json [hunk 2]
```diff
@@ -234,15 +234,15 @@
...
-              },
               "Operation-Location": {
                 "type": "string",
                 "format": "uri",
                 "description": "The location of an instance of AnalyzeResultOperation"
+              },
+              "Retry-After": {
+                "type": "integer",
+                "format": "int32",
+                "description": "The Retry-After header can indicate how long the client should wait before polling the operation status."
```
[the remark this answers is not in the document]
Comment (timotheeguerin) [reply specification/ai/data-plane/DocumentIntelligence/preview/2024-02-29-preview/DocumentIntelligence.json:245]: We did except for response headers which was fixed in this release

**Review by markcowl — APPROVED** [review]

**Pull request merged**